### DISCUSSION TITLE
Ruff v0.4.5 (Integrated LSP)

### DISCUSSION THREAD

**Discussion by S1M0N38** [LazyVim/LazyVim#3318] [General]
With v0.4.5, `ruff` is now shipped with LSP, and the plan is to completely replace `ruff-lsp`.
Is there any plan to migrate extra.lang.python to this new LSP client?
Maybe dropping `pyright` if some must-have features like `hover` are supported?

See https://astral.sh/blog/ruff-v0.4.5

**Comment by dpetka2001** [comment]
In your `/lua/config/options.lua` you can use `vim.g.lazyvim_python_ruff = "ruff"` to use ruff instead of ruff_lsp. For the moment, I believe it will stay as is, at least until ruff LSP server gets more stable and feature-rich. That's why this option was added, so users can choose on their own. It was temporarily changed to use ruff instead of ruff_lsp and some users mentioned some issues, so it was changed to use by default ruff_lsp, while giving users the option to opt-in into ruff if they choose so.
